scc/cert/serialization/types.go

```diff
@@ -0,0 +1,111 @@
+package serialization
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// HexBytes represents a byte slice that is serialized as a hexadecimal string with a "0x" prefix.
+type HexBytes []byte
+
+// MarshalJSON converts the HexBytes into a JSON-compatible hex string with a "0x" prefix.
+func (h HexBytes) MarshalJSON() ([]byte, error) {
+	if h == nil {
+		return []byte("null"), nil
+	}
+	return []byte(fmt.Sprintf("\"0x%x\"", []byte(h))), nil
+}
+
+// UnmarshalJSON parses a JSON hex string into a HexBytes slice.
+// The input string must have a "0x" prefix and be an even-length hex string.
```

Comment:
This is actually not true. The implementation accepts odd-length inputs, as well as the special case of a "null" value.